### PR TITLE
Create 'docs as code' pattern

### LINKED ISSUE #146: Create 'docs-as-code' pattern

### PR DESCRIPTION
**Pull request by edhamiltonHO**
# Content change 
I can confirm:
- [x] Content does not include any code or configuration changes (excluding frontmatter information)
- [x] Content meets the content standards
e.g. [Writing a principle](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-principle/) and [Writing a standard](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-standard/)
- [x] Content is suitable to open source, i.e.:
    - Content does not relate to unreleased gov policy
    - Content does not refer to anti-fraud mechanisms
    - Content does not include sensitive business logic

Will close #146

### CODE REVIEW

**Comment by RobJohnson-HO** [comment]
[sad]   Robert Johnson reacted to your message:
________________________________

File: docs/patterns/docs-as-code.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+{% import "macros/tags.njk" as pageTag %}
+
+Last updated: {{ page.date | postDate }}
+{{ pageTag.tags(tags)  }}
+
+The Home Office [DDaT Strategy 2024 - Principle 2 - Create shared technology products](https://www.gov.uk/government/publications/home-office-digital-data-and-technology-strategy-2024/home-office-digital-data-and-technology-strategy-2024#create-shared-technology-products) provides direction for improved developer documentation.  
+
+> "We will implement a [docs as code approach](https://www.gov.uk/guidance/how-to-document-apis) to documentation to ensure it develops in tandem with products, and to integrate technical writers with development and operations teams."
```
Comment (edhamiltonHO) [inline docs/patterns/docs-as-code.md:19]:
```suggestion
> "We will implement a docs as code approach to documentation to ensure it develops in tandem with products"
```

File: docs/patterns/docs-as-code.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+The Home Office software engineering Ways of Working guild has investigated some different technical products for implementing docs as code approaches. An outline of options is below, this site uses the Eleventy plugin.
+
+#### Technical Options
+
+| Option and links to documentation | Reviewed | Recommended for use? | Comments | Examples of HO documentation using this option |
+|---|---|---|---|---|
+| Middle man and GDS Tech Docs Template <br><br> https://middlemanapp.com/ <br><br> https://github.com/alphagov/tech-docs-template | YES | YES | Ruby based. <br><br> The original GDS sponsored effort to create a gov.uk compliant docs as code pattern. | https://ho-cto.github.io/sre-monitoring-as-code/ |
+| X-gov Eleventy Plugin <br><br> https://x-govuk.github.io/posts/govuk-eleventy-plugin/ | YES | YES | Node.js based. <br><br> A newer offering that also provides gov.uk styling and easy to configure search functionality. | https://ukhomeoffice.github.io/hocs/get-started/ |
+| Hugo <br><br> https://gohugo.io/getting-started/quick-start/ | YES | NO | Go based. <br><br> Static site generator without existing pattern for creating gov.uk or Home Office styled pages. | N/A |
```
Comment (robertdeniszczyc2) [inline docs/patterns/docs-as-code.md:61]:
```suggestion
#### Technical Options

| Option and links to documentation | Reviewed | Comments | Examples of HO documentation using this option |
|---|---|---|---|
| Middle man and GDS Tech Docs Template <br><br> https://middlemanapp.com/ <br><br> https://github.com/alphagov/tech-docs-template | YES | Ruby based. <br><br> The original GDS sponsored effort to create a gov.uk compliant docs as code pattern. | https://ho-cto.github.io/sre-monitoring-as-code/ |
| X-gov Eleventy Plugin <br><br> https://x-govuk.github.io/posts/govuk-eleventy-plugin/ | YES | Node.js based. <br><br> A newer offering that also provides gov.uk styling and easy to configure search functionality. | https://ukhomeoffice.github.io/hocs/get-started/ |
```

File: docs/patterns/docs-as-code.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+A docs as code approach also gives you the following benefits:
+
+- Writers integrate better with development teams
+- Developers will often write a first draft of documentation, reducing the time writers spend interviewing developers for technical accuracy
+- You can block merging of new features if they do not include documentation, which incentivises developers to write about features while they are fresh
+- Allows for easy iteration improvement of content inline with feature changes and feedback
+
+### Getting started
+
+The Home Office software engineering Ways of Working guild has investigated some different technical products for implementing docs as code approaches. An outline of options is below, this site uses the Eleventy plugin.
```
Comment (jeff-horton-ho-sas) [inline docs/patterns/docs-as-code.md:48]:
```suggestion
The Home Office software engineering Ways of Working guild has investigated some different technical products for implementing docs as code approaches. An outline of options is below, this site uses the x-gov Eleventy plugin.
```

**Review by jeff-horton-ho-sas — APPROVED** [review]

**Pull request merged**